sdk/iOS/src/MSCoreDataStore.m

```diff
@@ -55,47 +55,40 @@ -(id) getRecordForTable:(NSString *)table itemId:(NSString *)itemId asDictionary
     
     fr.predicate = [NSPredicate predicateWithFormat:@"%K ==[c] %@", MSSystemColumnId, itemId];
     
-    if (asDictionary) {
-        fr.resultType = NSDictionaryResultType;
-    }
-    
     NSArray *results = [self.context executeFetchRequest:fr error:error];
     if (!results || (error && *error)) {
         return nil;
     }
     
-    NSDictionary *item = [results firstObject];
-    if (asDictionary) {
-        // Otherwise this dictionary can't be worked with as user's expect
-        if (item) {
-            return [NSDictionary dictionaryWithDictionary:item];
-        } else {
-            return nil;
-        }
+    NSManagedObject *item = [results firstObject];
+    
+    if (item && asDictionary) {
+        
+        NSDictionary *result = [item dictionaryWithValuesForKeys:nil];
+
+        // The type of |result| is |NSKnownKeysDictionary|, an undocumented subclass of |NSMutableDictionary|.
+        // For it to work like a regular |NSMutableDictionary| we need to copy the contents into an
+        // |NSMutableDictionary| instance OR into an |NSDictionary| instance and then make a mutable copy.
+        return [NSDictionary dictionaryWithDictionary:result];
     }
     
-    return [results firstObject];
+    return item;
 }
 
 +(NSDictionary *) tableItemFromManagedObject:(NSManagedObject *)object
```

Comment:
Leaving this as a class method to avoid a breaking change in the dev branch. We can change this to an instance method once we integrate these changes to MobileApp.